kf_lib_data_ingest/etl/ingest_pipeline.py

```diff
@@ -23,97 +22,98 @@
 
-        :param target_api_config_path: Path to the target api config file
-        :param auto_transform: Boolean specifies whether to use automatic
-        graph-based transformation or user guided transformation
-        :param use_async: Boolean specifies whether to do ingest
-        asynchronously or synchronously
-        :param target_url: URL of the target API, into which data will be
-        loaded. Use default if none is supplied
-        :param log_level_name: case insensitive name of log level
-        (i.e. info, debug, etc) to control logging output.
-        See keys in logging._nameToLevel dict for all possible options
-        """
-        # Get args, kwargs
-        frame = inspect.currentframe()
-        args, _, _, values = inspect.getargvalues(frame)
-        kwargs = {arg: values[arg] for arg in args[2:]}
+        self.target_api_config_path = target_api_config_path
+        self.auto_transform = auto_transform
+        self.use_async = use_async
+        self.target_url = target_url
 
         # Get log params from dataset_ingest_config
-        log_dir = self.data_ingest_config.log_dir
+        log_dir = log_dir or self.data_ingest_config.log_dir
         log_kwargs = {param: getattr(self.data_ingest_config, param)
                       for param in ['overwrite_log', 'log_level']}
 
+        if overwrite_log is not None:
+            log_kwargs['overwrite_log'] = overwrite_log
+
         # Apply any log parameter overrides
-        log_level_name = kwargs.pop('log_level_name')
         log_level = logging._nameToLevel.get(str(log_level_name).upper())
         if log_level:
             log_kwargs['log_level'] = log_level
 
         # Setup logger
-        setup_logger(log_dir, **log_kwargs)
-        self.logger = logging.getLogger(__name__)
-
-        # Log the start of the run with ingestion parameters
-        run_msg = ('BEGIN data ingestion.\n\t-- Ingestion params --\n\t{}'
-                   .format(pformat(kwargs)))
-        self.logger.info(run_msg)
-
-        # Top level exception handler
-        # Catch exception, log it to file and console, and exit
-        try:
-            self._run(target_api_config_path, **kwargs)
-        except Exception as e:
-            msg = str(e) + '\nExiting.'
-            logging.exception(e)
-            exit(1)
-
-        # Log the end of the run
-        self.logger.info('END data ingestion')
+        self.log_file_path = setup_logger(log_dir, **log_kwargs)
+        self.logger = logging.getLogger(type(self).__name__)
 
-    def _run(self, target_api_config_path, auto_transform=False,
-             use_async=False, target_url=DEFAULT_TARGET_URL):
-        """
-        Runs the ingest pipeline
+        # Log args, kwargs
+        frame = inspect.currentframe()
+        args, _, _, values = inspect.getargvalues(frame)
+        kwargs = {arg: values[arg] for arg in args[2:]}
+        self.logger.info(
```

Comment:
Sorry another weird thing I did in the early days.. 

Removed the `\t`, it was making the kwargs look weird
```suggestion
        self.logger.info(f'-- Ingest Params --\n{pformat(kwargs)}')
```

Reply:
👍